opencdx-health/src/main/java/cdx/opencdx/health/model/OpenCDXAnalysisEngineModel.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2024 Safe Health Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cdx.opencdx.health.model;
+
+import cdx.opencdx.commons.data.OpenCDXIdentifier;
+import cdx.opencdx.grpc.data.AnalysisEngine;
+import com.google.protobuf.Timestamp;
+import java.time.Instant;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.annotation.*;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+/**
+ * Model for Analysis Engine in Mongo. Features conversions to/from Protobuf messages.
+ */
+@Slf4j
+@Data
+@Builder
+@AllArgsConstructor
+@RequiredArgsConstructor
+@Document("analysis-engine")
+@SuppressWarnings({"java:S3776", "java:S1117", "java:S116"})
+public class OpenCDXAnalysisEngineModel {
+    @Id
+    private OpenCDXIdentifier id;
+
+    @Version
+    private long version;
+
+    private String name;
+
+    private String organizationId;
```

Comment:
This should be an OpenCDXIdentifier not a string.

Reply:
Done